agent/internal/taskoutput/test_log.go

```diff
@@ -166,25 +220,62 @@ func (h *testLogDirectoryHandler) ingest(ctx context.Context, path string) {
 		}
 	}()
 
-	sender, err := h.createSender(ctx, logPath)
+	sender, err := h.createSender(ctx, logPath, sequence)
 	if err != nil {
 		h.logger.Task().Error(errors.Wrapf(err, "creating Sender for test log '%s'", path))
 		return
 	}
 
+	// Jump to the specified offset in the file.
+	if offset > 0 {
+		_, err := f.Seek(offset-1, io.SeekStart)
+		if err != nil {
+			h.logger.Task().Error(errors.Wrapf(err, "seeking offset for test log '%s'", path))
+			return
+		}
+	}
 	r := bufio.NewReader(f)
-	for {
-		line, err := r.ReadString('\n')
+
+	// If the offset is not at the beginning of a line, skip to the next
```

Comment:
I think L239-L260 can be refactored into a helper for readability, something like `getInitialPosition()`

Reply:
Same as above for the `run` function, breaking it up made it feel less readable. If you feel strongly about it, I can break it up or add more comments.